pymc_marketing/hsgp_kwargs.py

```diff
@@ -0,0 +1,80 @@
+#   Copyright 2024 The PyMC Labs Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
```

Comment:
Top line docstring for the documentation

Remind me why this became a new module? It didn't import correctly when in the tvp.py module?

Reply:
When it was in the `tvp.py` module there was as circular support error

Reply:
Ah. Could it be resolved with from __future__ import Annotations?

Reply:
Thanks for the tip! I was not able to make it work as it is just not a type hint but we check the instance of HSGPKwargs